mllib/src/main/scala/org/apache/spark/mllib/optimization/LBFGS.scala

```diff
@@ -60,6 +60,8 @@ class LBFGS(private var gradient: Gradient, private var updater: Updater)
   /**
    * Set the convergence tolerance of iterations for L-BFGS. Default 1E-4.
    * Smaller value will lead to higher accuracy with the cost of more iterations.
+   * This value must be nonnegative. Lower convergence tolerance is less tolerant
```

Comment:
Nit: works better as "Lower convergence values are less tolerant and therefore ..." as you have elsewhere. (I can get to this on merge if you don't first.) LGTM otherwise. I think we can ignore the Jenkins timeout. I will verify the docs build before merging.